cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -162,9 +162,7 @@ void MapBuilderBridge::RunFinalOptimization() {
 
 bool MapBuilderBridge::SerializeState(const std::string& filename,
                                       const bool include_unfinished_submaps) {
-  cartographer::io::ProtoStreamWriter writer(filename);
-  map_builder_->SerializeState(include_unfinished_submaps, &writer);
-  return writer.Close();
+  map_builder_->SerializeStateToFile(include_unfinished_submaps, filename);
```

Comment:
This needs to return a boolean indicating the success of the serialization. I'll do a PR in cartographer to allow implementing this in a proper way.

Reply:
With https://github.com/googlecartographer/cartographer/pull/1428 you can just do `return map_builder_->SerializeStateToFile(include_unfinished_submaps, filename);`